### DISCUSSION TITLE
Add music-video as a medium option

### DISCUSSION THREAD

**Discussion by mitchdowney** [Podcastindex-org/podcast-namespace#581] [Ideas]
I think a music video is a categorically different type of medium experience than audio-only music. Music videos are also a more specific type of experience than "videos" broadly. What do you think about adding music-video as a medium option?

**Comment by thomasrynne** [comment]
Interesting thought. I guess the question is, what's the difference between a medium='music-video' feed with video enclosures and a medium='music' feed with video enclosures?

**Reply by mitchdowney** [reply]
It's a good question...we do however already have a medium tag of `video` to declare a whole feed a "video feed", but a feed could also contain a combination of audio and video, in which case the feed could have the `mixed` medium.

My understanding is the `medium` tag is designed to easily signal to apps what type of UX they should use when playing content from an RSS feed. We can infer this in some cases from things like `audio` or `video` from enclosure tags, but that is less explicit than setting a `medium` tag.

So, if we're going to have a `video` medium, and a `film` medium, at the moment I'm not seeing why wouldn't also want a `music-video` medium for convenience.

**Reply by ryan-lp** [reply]
Another thought is that musicians aren't going to produce a music video for every track in an album, but may still want to keep the whole album, both the ones with and without videos, in the same "album" feed. And then the app may make UX adjustments based on whether the track has a video.

So the behaviour of the `music` medium alone is likely to evolve into something where apps will make UX adjustments if a particular item comes with a video. And then if an entire `music` feed consists of items with videos, it would effectively give the same UX as if the feed had the `music-video` medium.

**Reply by mitchdowney** [reply]
Ah hmm...I hadn't thought of it that way. It still seems a bit odd to me that we would have a `video` medium tag, if apps are expected to rely on the enclosures to indicate if something is a video or not. It seems like there could be 3 types of music feeds then, `music`, `music-video`, and `music-mixed`. I don't want to insist on so many types of mediums though.

I'm not sure if I should close the discussion at this point. The probability of people sharing music videos in the near future seems low (since video itself is still pretty rare), so maybe this is something we just revisit if it seems like there's a need later.

**Comment by redimongo** [comment]
What does this have to do with podcasts and the podcast 2.0 standard?

**Reply by mitchdowney** [reply]
My understanding is Podcasting 2.0 views podcasting as encompassing all RSS-based audio/video content distribution. For example, it's just as valid to share a music album in an RSS feed as it is to share a talk show in an RSS feed. Podcasting is also not limited to audio, as many video podcasts exist. Therefore, music videos can be included in the P2.0 `medium` standard, although I don't know if a `medium="music-video"` tag is a good solution for it.

**Comment by agates** [comment]
I think we should add this as `musicVideo` and `musicVideoL` simply because it's easy to see an application wanting to care only about music videos, nothing more nothing less.

I was considering using categories to classify videos as music, but that doesn't solve the is-about problem where videos could be about music.

It's easy to see someone wanting an MTV style application and I think that use case as a paradigm has historical merit.

cc @daveajones

**Comment by jamescridland** [comment]
medium="music"

Use the filetypes to work out if it's a video or an audio file.

Particularly, musicians may add an alternate enclosure of a music video,
associated with audio of the same track.

This is the same argument as the nonsensical medium="video"

**Reply by daveajones** [reply]
I created a thread to hash out the medium="video" issue:  #586

**Comment by daveajones** [comment]
`medium` is a channel level tag.  Part of the recent confusion about medium="video" is because of the conceptual munging of enclosure file types with channel mediums.  The medium declares what the entire feed is, like audiobook.  The media served in the enclosures will be expected to adhere to that concept (items become audiobook chapters).

With that idea in mind, do we anticipate there being a feed comprised entirely of music videos?

**Reply by mitchdowney** [reply]
Given that feeds often represent albums currently, I would think there may be feeds that contain all music videos for an album, or a feed with all of the music videos for an artist? Also, couldn't musicians combine music videos along with audio music files into a single feed? What should those hybrid music feeds be labeled? "video" or "music" or "musicVideo"?

I'm fine with just closing this topic and waiting until music videos are a thing in podcasting 2.0. I just partially wanted to bring it up because I think there may be debate over whether a music video feed should be `medium="video"` or `medium="music"`, and it would nice to be able to provide clear guidance on what to use when and why, if not create a new `musicVideo` tag.

**Reply by daveajones** [reply]
All valid questions.  Asking these sorts of questions is the only way we end up making good things.

**Reply by mitchdowney** [reply]
Thinking over this more...even though an artist _can_ combine music videos and an audio album into the same feed, maybe that's not something we recommend? It doesn't make sense as a listener to sit in front of a laptop watching a music video, then an audio track comes on, then you...look away and do something else? Maybe hybrid medium feeds, while possible with RSS feeds, should be discouraged. Clearer separation of concerns would be better for UX and searchability. That said, while it might make sense to separate music videos from audio music tracks in RSS feeds, I don't think we want podcasters to interpret that as they should also separate their video and audio RSS feeds into `medium="video"` and `medium="podcast"` feeds? Or do we?

At the moment I'm inclined to say as a _recommendation_, that if a podcast has video and audio in it, that it should "round up" to the more multi-sensory medium type, so declare itself the `video` medium.  BUT, apps still need a way to know that this feed, thanks to alternateEnclosures, is not just a video, but a "podcast" that can be played in audio only podcast apps. It seems like there is another layer of tagging / keywords / hashtags / something that may be needed so apps can filter on cross-cutting medium categories.

Or, we encourage podcasters to create a separate feed for each medium type. "Create a feed for podcast apps. Create a feed for video apps. Create a feed for music apps. Create a feed for music video apps." The content of these may overlap, but apps can select which feed is more suitable for its platform. Some people may fall out of there chair at the suggestion of more RSS feed segmentation 😅 I think the pros and cons are worth considering though.

**Reply by agates** [reply]
The question if there could be an app dedicated toward Music videos.  As there is historical precedent here, the answer is yes.

And for that use case, medium=video with a category indicating music isn't enough because some videos will be about music and not be music.
